File: convert.js
const fs = require('fs');

const content = fs.readFileSync('apps/web/public/mapping.html', 'utf8');

// Extract CSS
const cssMatch = content.match(/<style>([\s\S]*?)<\/style>/);
const cssContent = cssMatch ? cssMatch[1].trim() : '';

// Extract JS
const jsMatch = content.match(/<script>([\s\S]*?)<\/script>/);
let jsContent = jsMatch ? jsMatch[1].trim() : '';
jsContent = jsContent.replace(
  "window.addEventListener('DOMContentLoaded',()=>{\n  loadLocateGrave();\n  setTimeout(initPreviews,120);\n  if(LOCATE_GRAVE){\n    setTimeout(()=>{ openView('public'); },400);\n  }\n});",
  "window.initMapping = () => {\n  loadLocateGrave();\n  setTimeout(initPreviews,120);\n  if(LOCATE_GRAVE){\n    setTimeout(()=>{ openView('public'); },400);\n  }\n};"
);
jsContent += '\n// Make global functions explicit\n';
['openView', 'closeView', 'setCamera', 'toggleLayer', 'flyToTarget', 'showGravePopup', 'closeGravePopup'].forEach(fn => {
  jsContent += `\nwindow.${fn} = ${fn};`;
});

// Extract Body
const bodyMatch = content.match(/<body>([\s\S]*?)<script/);
const bodyHtml = bodyMatch ? bodyMatch[1].trim() : '';

// Clean up HTML for React dangerouslySetInnerHTML
const pageTsx = `'use client';

import { useEffect, useRef } from 'react';
import Script from 'next/script';
import './mapping.css';

export default function MappingPage() {
  const initialized = useRef(false);

  useEffect(() => {
    const timer = setInterval(() => {
      if (typeof window !== 'undefined' && (window as any).initMapping && !initialized.current) {
        (window as any).initMapping();
        initialized.current = true;
        clearInterval(timer);
      }
    }, 100);
    return () => clearInterval(timer);
  }, []);

  return (
    <div style={{ width: '100%', height: 'calc(100vh - 60px)', position: 'relative' }}>
      <div dangerouslySetInnerHTML={{ __html: \`${bodyHtml.replace(/`/g, '\\`')}\` }} />
      <Script src="https://cdnjs.cloudflare.com/ajax/libs/three.js/r128/three.min.js" strategy="beforeInteractive" />
      <Script src="/mapping-script.js" strategy="afterInteractive" />
    </div>
  );
}
`;

fs.writeFileSync('apps/web/app/(user)/mapping/mapping.css', cssContent, 'utf8');
fs.writeFileSync('apps/web/public/mapping-script.js', jsContent, 'utf8');
fs.writeFileSync('apps/web/app/(user)/mapping/page.tsx', pageTsx, 'utf8');

console.log('Success');

File: convert.py
import re

with open('apps/web/public/mapping.html', 'r', encoding='utf-8') as f:
    content = f.read()

# Extract CSS
css_match = re.search(r'<style>(.*?)</style>', content, re.DOTALL)
css_content = css_match.group(1).strip()

# Extract JS
js_match = re.search(r'<script>(.*?)</script>', content, re.DOTALL)
js_content = js_match.group(1).strip()
js_content = js_content.replace("window.addEventListener('DOMContentLoaded',()=>{\n  loadLocateGrave();\n  setTimeout(initPreviews,120);\n  if(LOCATE_GRAVE){\n    setTimeout(()=>{ openView('public'); },400);\n  }\n});", "window.initMapping = () => {\n  loadLocateGrave();\n  setTimeout(initPreviews,120);\n  if(LOCATE_GRAVE){\n    setTimeout(()=>{ openView('public'); },400);\n  }\n};")
js_content += '\n// Make global functions explicit\n'
for fn in ['openView', 'closeView', 'setCamera', 'toggleLayer', 'flyToTarget', 'showGravePopup', 'closeGravePopup']:
    js_content += f'\nwindow.{fn} = {fn};'

# Extract Body
body_match = re.search(r'<body>(.*?)<script', content, re.DOTALL)
body_html = body_match.group(1).strip()

# Clean up HTML for React dangerouslySetInnerHTML
page_tsx = f"""'use client';

import {{ useEffect, useRef }} from 'react';
import Script from 'next/script';
import './mapping.css';

export default function MappingPage() {{
  const initialized = useRef(false);

  useEffect(() => {{
    const timer = setInterval(() => {{
      if (typeof window !== 'undefined' && (window as any).initMapping && !initialized.current) {{
        (window as any).initMapping();
        initialized.current = true;
        clearInterval(timer);
      }}
    }}, 100);
    return () => clearInterval(timer);
  }}, []);

  return (
    <div style={{{{ width: '100%', height: 'calc(100vh - 60px)', position: 'relative' }}}}>
      <div dangerouslySetInnerHTML={{{{ __html: `{body_html.replace('`', '\\\\`')}` }}}} />
      <Script src="https://cdnjs.cloudflare.com/ajax/libs/three.js/r128/three.min.js" strategy="beforeInteractive" />
      <Script src="/mapping-script.js" strategy="afterInteractive" />
    </div>
  );
}}
"""

with open('apps/web/app/(user)/mapping/mapping.css', 'w', encoding='utf-8') as f:
    f.write(css_content)

with open('apps/web/public/mapping-script.js', 'w', encoding='utf-8') as f:
    f.write(js_content)

with open('apps/web/app/(user)/mapping/page.tsx', 'w', encoding='utf-8') as f:
    f.write(page_tsx)

print('Success')
